test/integration-test/src/utils.rs

```diff
@@ -0,0 +1,61 @@
+//! Utilities to run tests
+
+use std::{
+    ffi::CString,
+    os::fd::RawFd,
+    sync::atomic::{AtomicU64, Ordering},
+};
+
+use aya::netlink_set_link_up;
+use libc::if_nametoindex;
+use netns_rs::{get_from_current_thread, NetNs};
+
+pub struct Netns {
+    pub name: String,
+    pub old_ns: NetNs,
+    pub ns: Option<NetNs>,
+}
+
+impl Netns {
+    pub fn new() -> Self {
+        static COUNTER: AtomicU64 = AtomicU64::new(0);
+        let pid = std::process::id();
+        let name = format!("aya-test-{pid}-{}", COUNTER.fetch_add(1, Ordering::Relaxed));
+
+        // Create and enter netns
+        let ns = NetNs::new(&name).unwrap_or_else(|e| panic!("Failed to create netns {name}: {e}"));
+        ns.enter().expect("Failed to enter network namespace");
+        let netns = Self {
+            old_ns: get_from_current_thread().expect("Failed to get current netns"),
+            ns: Some(ns),
+            name,
+        };
+
+        // By default, the loopback in a new netns is down. Set it up.
+        let lo = CString::new("lo").unwrap();
+        unsafe {
+            let idx = if_nametoindex(lo.as_ptr());
+            if idx == 0 {
+                panic!("Interface `lo` not found in netns {}", netns.name);
```

Comment:
> On success, if_nametoindex() returns the index number of the
       network interface; on error, 0 is returned and [errno](https://man7.org/linux/man-pages/man3/errno.3.html) is set to
       indicate the error.

please emit the value of [`std::io::Error::last_os_error`](https://doc.rust-lang.org/stable/std/io/struct.Error.html#method.last_os_error)

Reply:
Done in 8ec9f73